tests/datatype/tests_HexBinary.cpp

```diff
@@ -0,0 +1,87 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+
+#include <doctest/doctest.h>
+#include <rdf4cpp/rdf.hpp>
+
+using namespace rdf4cpp::rdf::datatypes::xsd;
+using rdf4cpp::rdf::datatypes::registry::HexBinaryRepr;
+
+TEST_SUITE("xsd:hexBinary") {
+    TEST_CASE("canonical") {
```

Comment:
Is canonical the right name for the testcases? some seem quite non-canonical (see above)